src/Compilers/Core/Portable/SourceGeneration/GeneratorContexts.cs

```diff
@@ -61,6 +62,11 @@ internal GeneratorExecutionContext(Compilation compilation, ParseOptions parseOp
         /// </summary>
         public ISyntaxReceiver? SyntaxReceiver { get; }
 
+        /// <summary>
+        /// If the generator registered an <see cref="ISyntaxReceiverWithContext"/> during initialization, this will be the instance created for this generation pass.
+        /// </summary>
+        public ISyntaxReceiverWithContext? SyntaxReceiverWithContext { get; }
```

Comment:
We can't have a single field without breaking back compat. An alternative approach could be to add a method to the context such as `GetReceiver<T>() where T : ISyntaxReceiverBase`

Reply:
Couldn't we have a single field and just implement each property `get` as the cast operation?

Not sure it's worth the trade off though.

Reply:
Yeah, we'd still need two public properties though, so its ergonomically not really any different for the end user.

---
In reply to: [562023044](https://github.com/dotnet/roslyn/pull/50419#discussion_r562023044) [](ancestors = 562023044)